### PR TITLE
8 trafficcounter raspberrypi

### LINKED ISSUE #8: TrafficCounter (raspberrypi)
- read ir sensor movement
- send reading to mqtt topic (1 topic per sensor / topic should be configurable for each sensor)

### PR DESCRIPTION
**Pull request by Solargi**
reading both VCNL 4010 and IR Obstacle Sensor each with his car counter which is publish on the respective topics.
beware on how to correctly connect the VCNL to the Grove shield, the instruction are in the code initial note, the pin are defined in the code.

some configurable parameter (ex: protocol, sensor ids,..) are defined in the code, next to their definition there is the commented code that will take the values from a config file, once a definitive and correct config file is defined, just substitute the parameter with the commented part and update the fields accordingly to set the value from the config file.

the file also contains some print statement that are useful for debugging purposes

### CODE REVIEW

File: traffic/TrafficCounter.py [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+
+#config connection
+broker_address=str(config['connection']['broker_address'])
+port = int(config['connection']['port'])
+username = str(config['connection']['username'])
+password = str(config['connection']['password'])
+##authentication object needed to publish on topic
+auth = {'username':username, 'password':password}
+##topic base config
+protocol = "json" ## str(config['connection']['protocol'])
```
Comment (chicco785) [inline traffic/TrafficCounter.py]: load form configuration

File: traffic/TrafficCounter.py [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+service_api_key = "test_key"#str(config['Street 2']['service_api_key'])
```
Comment (chicco785) [inline traffic/TrafficCounter.py]: load form configuration

File: traffic/TrafficCounter.py [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+# ir_car_counter = 0
+
+
+
+
+## Obstacle sensor data and function
+## connect the obstacle sensor to port d8
+os_pin = 8
+grovepi.pinMode(os_pin,"INPUT")
+os_id = "os" #str(config['Street 2']['os_id'])
```
Comment (chicco785) [inline traffic/TrafficCounter.py]: load form configuration (don't use spaces) so "street2"

File: traffic/TrafficCounter.py [hunk 4]
```diff
@@ -0,0 +1,132 @@
...
+auth = {'username':username, 'password':password}
+##topic base config
+protocol = "json" ## str(config['connection']['protocol'])
+service_api_key = "test_key"#str(config['Street 2']['service_api_key'])
+
+##ir Sensor data and functions
+##Connect the ir sensor at port d6 NOT d7 (see initial note)
+ir_pin = 7 #str(config['Street 1']['ir_pin'])
+grovepi.pinMode(ir_pin,"INPUT")
+ir_id = "ir" #str(config['Street 1']['ir_id'])
```
Comment (chicco785) [inline traffic/TrafficCounter.py]: load form configuration (don't use spaces) so "street1"

File: traffic/TrafficCounter.py [hunk 5]
```diff
@@ -0,0 +1,132 @@
...
+password = str(config['connection']['password'])
+##authentication object needed to publish on topic
+auth = {'username':username, 'password':password}
+##topic base config
+protocol = "json" ## str(config['connection']['protocol'])
+service_api_key = "test_key"#str(config['Street 2']['service_api_key'])
+
+##ir Sensor data and functions
+##Connect the ir sensor at port d6 NOT d7 (see initial note)
+ir_pin = 7 #str(config['Street 1']['ir_pin'])
```
Comment (chicco785) [inline traffic/TrafficCounter.py]: load form configuration (don't use spaces) so "street1"

File: traffic/TrafficCounter.py (hunk 1 above)
Comment (Solargi) [reply traffic/TrafficCounter.py]: done config.ini was defind in issue #19 
the parameters username an password used to define auth are taken from the config.ini file

File: traffic/TrafficCounter.py (hunk 2 above)
Comment (Solargi) [reply traffic/TrafficCounter.py]: done config.ini was defind in issue #19 the parameters username an password used to define auth are taken from the config.ini file

File: traffic/TrafficCounter.py (hunk 3 above)
Comment (Solargi) [reply traffic/TrafficCounter.py]: done config.ini was defind in issue #19

File: traffic/TrafficCounter.py (hunk 4 above)
Comment (Solargi) [reply traffic/TrafficCounter.py]: done config.ini was defind in issue #19

File: traffic/TrafficCounter.py (hunk 5 above)
Comment (Solargi) [reply traffic/TrafficCounter.py]: done config.ini was defind in issue #19

File: traffic/TrafficCounter.py [hunk 6]
```diff
@@ -0,0 +1,133 @@
...
+# config connection
+broker_address = str(config['connection']['broker_address'])
+port = int(config['connection']['port'])
+username = str(config['connection']['username'])
+password = str(config['connection']['password'])
+# authentication object needed to publish on topic
+auth = {'username': username, 'password': password}
+# topic base config
+protocol = str(config['connection']['protocol'])
+service_api_key = str(config['Traffic_counter']['traffic_service_api_key'])
```
Comment (chicco785) [inline traffic/TrafficCounter.py]: let's be consistent Traffic_counter -> traffic_counter

File: traffic/TrafficCounter.py [hunk 7]
```diff
@@ -0,0 +1,133 @@
...
+
+# VCNL 4010 IR Sensor setup
+# Connect the ir sensor at port d6 NOT d7 (see initial note)
+ir_pin = int(config['Traffic_counter']['ir_pin'])
```
Comment (chicco785) [inline traffic/TrafficCounter.py]: let's be consistent Traffic_counter -> traffic_counter

File: traffic/TrafficCounter.py [hunk 8]
```diff
@@ -0,0 +1,133 @@
...
+grovepi.pinMode(ir_pin, "INPUT")
+ir_id = str(config['Traffic_counter']['ir_id'])
```
Comment (chicco785) [inline traffic/TrafficCounter.py]: let's be consistent Traffic_counter -> traffic_counter

File: traffic/TrafficCounter.py [hunk 9]
```diff
@@ -0,0 +1,133 @@
...
+ir_topic = topic_constructor(protocol, service_api_key, ir_id)
+
+
+# Obstacle sensor set up
+# connect the obstacle sensor to port d8
+os_pin = int(config['Traffic_counter']['os_pin'])
+grovepi.pinMode(os_pin, "INPUT")
+os_id = str(config['Traffic_counter']['os_id'])
```
Comment (chicco785) [inline traffic/TrafficCounter.py]: let's be consistent Traffic_counter -> traffic_counter

File: traffic/TrafficCounter.py [hunk 10]
```diff
@@ -0,0 +1,133 @@
...
+# Connect the ir sensor at port d6 NOT d7 (see initial note)
+ir_pin = int(config['Traffic_counter']['ir_pin'])
+grovepi.pinMode(ir_pin, "INPUT")
+ir_id = str(config['Traffic_counter']['ir_id'])
+ir_topic = topic_constructor(protocol, service_api_key, ir_id)
+
+
+# Obstacle sensor set up
+# connect the obstacle sensor to port d8
+os_pin = int(config['Traffic_counter']['os_pin'])
```
Comment (chicco785) [inline traffic/TrafficCounter.py]: let's be consistent Traffic_counter -> traffic_counter

**Review by chicco785 — CHANGES_REQUESTED** [review]
see my comments,
and please upload an example config file with comments on variables

File: traffic/TrafficCounter.py (hunk 6 above)
Comment (Solargi) [reply traffic/TrafficCounter.py]: done

File: traffic/TrafficCounter.py (hunk 7 above)
Comment (Solargi) [reply traffic/TrafficCounter.py]: done

File: traffic/TrafficCounter.py (hunk 8 above)
Comment (Solargi) [reply traffic/TrafficCounter.py]: done

File: traffic/TrafficCounter.py (hunk 9 above)
Comment (Solargi) [reply traffic/TrafficCounter.py]: done

File: traffic/TrafficCounter.py (hunk 10 above)
Comment (Solargi) [reply traffic/TrafficCounter.py]: done

**Review by chicco785 — APPROVED** [review]

**Pull request merged**